### PR TITLE
Array - rotate matrix

### PR DESCRIPTION
**Pull request by DevratAgarwal**
## Checklist
- Solution is tested on sphere engine <br/>
    `Note:` Please copy only the solution part. Do not copy/paste the template code. That way, any issues with template code can be caught early on.
    - [ ] Author
    - [x] Reviewer
- PR title is correct - [PR title conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#title-for-a-pull-request)
    - [x] Author
    - [ ] Reviewer
- Folder/File naming is correct - [Naming conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#naming-of-files-and-folders)
    - [x] Author
    - [ ] Reviewer
- Coding conventions are followed - [Coding conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#coding-conventions)
    - [ ] Author
    - [x] Reviewer
- Test cases (action item for **reviewer**) <br/>
    - To be ensured by checking the problem on sphere-engine
    - If any test cases are missing, please add them - [how to add new test cases](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#adding-new-test-cases)
    - [x] There are at least 2 test cases
    - [x] There is at least 1 test case that is different from the sample test cases

### CODE REVIEW

**Review by rpapavan97 — APPROVED** [review]
1. Added ";" at the end of each line
2. Modified the logic to complete the rotating in one nested loop instead of two nested loops (previously)
3. Added spaces around the binary operators

**Comment by sailesh-ramayanam** [comment]
@rpapavan97 Output explanation uses the term `transpose`. We are not doing a transpose. Please remove the term.

File: solutions/arrays/rotate-matrix.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+function readLine() {
+  idx++;
+  return data[idx - 1];
+}
+
+// -------- Do NOT edit anything above this line ----------
+// Use readLine() for taking input, it will read one line of from the input  and is stored in string format
+function rotate(arr, numOfRows){
+	console.log(arr[0].length);
```
Comment (sailesh-ramayanam) [inline solutions/arrays/rotate-matrix.js:14]: @DevratAgarwal Please do not print inside this function. Read about single responsibility principle. Perhaps we can avoid writing a function.

**Review by sailesh-ramayanam — CHANGES_REQUESTED** [review]

**Comment by rpapavan97** [comment]
@sailesh-ramayanam updated the Explanation section in problem

**Comment by DevratAgarwal** [comment]
@rpapavan97 @sailesh-ramayanam changes are addressed

**Comment by prathyusha10x** [comment]
@rpapavan97 @sailesh-ramayanam please review the code and approve the PR by tomorrow EOD

File: solutions/arrays/rotate-matrix.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            let temporary = answer[row][column];
+            answer[row][column]=answer[row][answer[0].length-1-column];
+            answer[row][answer[0].length-1-column]=temporary;
+        }
+    }
+	return answer;
+}
+let val = parseInt(readLine());
+let lst = [];
+for(let row=0;row<val;row++)
```
Comment (sailesh-ramayanam) [inline solutions/arrays/rotate-matrix.js:39]: Rename `row` to `rowIndex`

File: solutions/arrays/rotate-matrix.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	for(let index=0;index<temporaryArray.length;index++)
+	{
+		temporaryArray[index]=parseInt(temporaryArray[index]);
+	}
+	lst.push(temporaryArray);
+    
+}
+console.log(lst[0].length);
+let out=rotate(lst)
+for(val of out){
```
Comment (sailesh-ramayanam) [inline solutions/arrays/rotate-matrix.js:51]: Rename `val` to `row`

File: solutions/arrays/rotate-matrix.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	}
+    for(let row=0;row<answer.length;row++)
+    {
+        for(let column=0;column<answer[0].length/2;column++)
+        {
+            let temporary = answer[row][column];
+            answer[row][column]=answer[row][answer[0].length-1-column];
+            answer[row][answer[0].length-1-column]=temporary;
+        }
+    }
```
Comment (sailesh-ramayanam) [inline solutions/arrays/rotate-matrix.js:34]: We don't need 2 sets of nested loops. We can do with 1 set.
```js
for colIndex from 0 to n-1{
    for rowIndex from n-1 to 0 {
         newRow.push(matrix[rowIndex][colIndex])
    }
    roatedMatrix.push(newRow)
}
```

File: solutions/arrays/rotate-matrix.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+data = data.split('\n');
+
+function readLine() {
+  idx++;
+  return data[idx - 1];
+}
+
+// -------- Do NOT edit anything above this line ----------
+// Use readLine() for taking input, it will read one line of from the input  and is stored in string format
+function rotate(lst)
```
Comment (sailesh-ramayanam) [inline solutions/arrays/rotate-matrix.js:13]: Please don't use a function. Write it as part of the main logic itself.

**Comment by DevratAgarwal** [comment]
@sailesh-ramayanam done all changes

**Review by sailesh-ramayanam — APPROVED** [review]

**Pull request merged**